### PR TITLE
Node changes

### PR DESCRIPTION
**Pull request by RosenKrumov**
## Changes
* Added first version of Node.
* Created classes for Node, Block and Transaction.
* Added genesis block and transaction.
* Added all http routes.

### CODE REVIEW

File: Node/noobcoin_node.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+var FAUCET_ADDRESS = '0x756F45E3FA69347A9A973A725E3C98bC4db0b5a0'; //TODO
+var FAUCET_PUBKEY = '0x756F45E3FA69347A9A973A725E3C98bC4db0b5a0'; //TODO
+var FAUCET_SIGNATURE = ['0x756F45E3FA69347A9A973A725E3C98bC4db0b5a0', '0x756F45E3FA69347A9A973A725E3C98bC4db0b5a0']; //TODO
+var INITIAL_COINS_DISTRIBUTION = 100000000;
+
+var node = {};
+var allTransactions = [];
+
+class Transaction {
+	constructor(fromAddress, toAddress, value, senderPublicKey, senderSignature, transactionHash, dateReceived, minedInBlockIndex, paid) {
```
Comment (presmihaylov) [inline Node/noobcoin_node.js]: signature can be split in several lines

File: Node/noobcoin_node.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+		this.senderSignature = senderSignature;
+		this.transactionHash = transactionHash;
+		this.dateReceived = dateReceived;
+		this.minedInBlockIndex = minedInBlockIndex;
+		this.paid = paid;	
+	}
+}
+
+class Block {
+	constructor(index, transactions, difficulty, previousBlockHash, minedBy, blockDataHash, nonce, dateCreated, blockHash) {
```
Comment (presmihaylov) [inline Node/noobcoin_node.js]: same here

File: Node/noobcoin_node.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+		}
+	});
+	
+	app.post('/blocks/notify', (req, res) => {
+	
+	});
+	
+	app.get('/peers', (req, res) => res.send(JSON.stringify(node.peers)));
+	app.post('/peers', (req, res) => {
+
```
Comment (presmihaylov) [inline Node/noobcoin_node.js]: add TODO

File: Node/noobcoin_node.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+		}
+	});
+	
+	app.get('/balance/:address/confirmations/:confirmations', (req, res) => {
+		var address = req.params.address;
+		var confirmations = req.params.number;
+	});
+	
+	app.post('/transactions/new', (req, res) => {
+
```
Comment (presmihaylov) [inline Node/noobcoin_node.js]: add TODO

**Review by presmihaylov — CHANGES_REQUESTED** [review]

File: Node/noobcoin_node.js (hunk 1 above)
Comment (RosenKrumov) [reply Node/noobcoin_node.js]: resolved

Comment (RosenKrumov) [reply Node/noobcoin_node.js]: done

**Review by presmihaylov — APPROVED** [review]

**Pull request merged**